include/h2o.h

```diff
@@ -467,6 +467,16 @@ enum {
 };
 
 /**
+ * holds various data related to the context
+ */
+typedef struct st_h2o_context_storage_item_t {
+    void (*dispose)(struct st_h2o_context_storage_item_t *self, h2o_context_t *ctx);
```

Comment:
How about _not_ passing `h2o_context_t`, considering the fact that it would already be partially destructed when the callback gets called?